contracts/PvpArena.sol

```diff
@@ -871,4 +844,49 @@ contract PvpArena is Initializable, AccessControlUpgradeable {
         //this is not final, but processing the loser will update the ranks leaving this player in the 4th position, which will be quickly replaced by other players
         processLoser(characterID);
     }
+
+    /// @dev distributes the ranking rewards pool to top players
+    // TODO: Re-evaluate if private or requires some permission
+    function _distributeSeasonRewards() private {
+        for (uint8 i = 0; i <= 15; i++) {
+            if (_fightersByTier[i].length() == 0) {
+                continue;
+            }
+            for (uint8 j = 0; j < _prizePercentages.length; j++) {
+                _transferRewards(
+                    _rankingByTier[i][j],
+                    j + 1,
+                    _rankingsPoolByTier[i]
+                );
+            }
+        }
+    }
+
+    /// @dev increases a players withdrawable funds depending on their position in the ranked leaderboard
+    function _transferRewards(
+        uint256 characterID,
+        uint8 position,
+        uint256 pool
+    ) private {
+        uint8 percentage = _prizePercentages[position];
+        address playerToTransfer = characters.ownerOf(characterID);
+        uint256 amountToTransfer;
+
+        amountToTransfer = (pool.mul(percentage)).div(100);
+
+        withdrawableFundsByAddress[
+            playerToTransfer
+        ] = withdrawableFundsByAddress[playerToTransfer].add(amountToTransfer);
+    }
+
+    /// @dev allows a player to withdraw their withdrawable funds
+    function withdraw() external {
```

Comment:
We also have a withdrawal function for pvpEarnings, so this should be more specific